strawberryfields/ops.py

```diff
@@ -867,8 +867,8 @@ def _apply(self, reg, backend, **kwargs):
         N = temp / pf.sqrt(2 * (1 + pf.cos(phi) * temp ** 4))
 
         # coherent states
-        c1 = (alpha ** l) / np.sqrt(ssp.factorial(l))
-        c2 = ((-alpha) ** l) / np.sqrt(ssp.factorial(l))
+        c1 = ((1.0 * alpha) ** l) / np.sqrt(ssp.factorial(l))
```

Comment:
```suggestion
        # Need to cast  alpha to float before exponentiation to avoid overflow
        c1 = ((1.0 * alpha) ** l) / np.sqrt(ssp.factorial(l))
```